src/Actions/ResolveForPropertyValueAction.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Spatie\Activitylog\Actions;
+
+class ResolveForPropertyValueAction
+{
+    /**
+     * Action that resolve property value of log
+     * that cannot be handled by PHP as default
+     *
+     * @param mixed $value
+     * @return mixed
+     */
+    public static function execute(mixed $value): mixed
+    {
+        $instance = new static;
+
+        /**
+         * Give a fallback value if value not a backed enum
+         */
+        if ($instance->isValueAreEnum($value)) {
+            return $value->value ?? $value->name;
+        }
+
+        return $value;
+    }
+
+    protected function isValueAreEnum($value): bool
```

Comment:
Sorry about that but "is value are enum" sounds strange - `isValueAnEnum()` would be more readable I think.
```suggestion
    protected function isValueAnEnum($value): bool
```

Reply:
Alright, great review